src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/SessionImpl.scala

```diff
@@ -168,36 +168,38 @@ private[omega_edit] class SessionImpl(p: Pointer, i: FFI) extends Session {
   }
 
   def search(
-      pattern: String,
+      pattern: Array[Byte],
       offset: Long,
       length: Option[Long] = None,
       caseInsensitive: Boolean = false,
       limit: Option[Long] = None
   ): List[Long] = {
-    val context =
-      i.omega_search_create_context(
-        p,
-        pattern,
-        0,
-        offset,
-        length.getOrElse(0),
-        caseInsensitive
-      )
-
-    try {
-      Iterator
-        .unfold(context -> 0) { case (context, numMatches) =>
-          Option.when(
-            limit.map(numMatches < _).getOrElse(true) && i
-              .omega_search_next_match(context, 1) > 0
-          )(
-            i.omega_search_context_get_offset(
-              context
-            ) -> (context, numMatches + 1)
-          )
-        }
-        .toList
-    } finally i.omega_search_destroy_context(context)
+    i.omega_search_create_context_bytes(
+      p,
+      pattern,
+      0,
```

Comment:
This is a problem.  You need to send in the pattern length explicitly here because otherwise Ωedit is going to try to do a strlen of that pattern which could contain null bytes or may not be terminated at all leading to a segmentation fault.